### PR TITLE
minor clean code

### PR DESCRIPTION
**Pull request by Marthyi**
- reduce verbosity of HttpUtilities

### CODE REVIEW

File: src/ReverseProxy.Core/Service/Proxy/HttpUtilities.cs [hunk 1]
```diff
@@ -13,55 +13,17 @@ internal static class HttpUtilities
...
-        }
+            string mth when HttpMethods.IsGet(mth) => HttpMethod.Get,
+            string mth when HttpMethods.IsPost(mth) => HttpMethod.Post,
+            string mth when HttpMethods.IsPut(mth) => HttpMethod.Put,
+            string mth when HttpMethods.IsDelete(mth) => HttpMethod.Delete,
+            string mth when HttpMethods.IsOptions(mth) => HttpMethod.Options,
+            string mth when HttpMethods.IsHead(mth) => HttpMethod.Head,
+            string mth when HttpMethods.IsPatch(mth) => HttpMethod.Patch,
+            string mth when HttpMethods.IsTrace(mth) => HttpMethod.Trace,
+            _ => throw new InvalidOperationException($"Unsupported request method '{method}'.")
```
Comment (Tratcher) [inline src/ReverseProxy.Core/Service/Proxy/HttpUtilities.cs:26]:
```suggestion
            // NOTE: Proxying "CONNECT" is not supported (by design!)
            _ => throw new InvalidOperationException($"Unsupported request method '{method}'.")
```

Comment (Marthyi) [reply src/ReverseProxy.Core/Service/Proxy/HttpUtilities.cs:27]: > 

- Good point, I 've re-inserted the comment.
- No issue detected on AtomicCounter. Here I just wanted to "force" the volatile access

**Comment by halter73** [comment]
> What was wrong with AtomicCounter? It doesn't look like you've changed the behavior, just the pattern?

I think I prefer the old way. The `volatile` keyword is too easy to miss.

**Comment by Tratcher** [comment]
Thanks

File: src/ReverseProxy.Core/Service/Proxy/HttpUtilities.cs [hunk 2]
```diff
@@ -13,55 +13,18 @@ internal static class HttpUtilities
...
+            string mth when HttpMethods.IsTrace(mth) => HttpMethod.Trace,
             // NOTE: Proxying "CONNECT" is not supported (by design!)
-            //if (HttpMethods.IsConnect(method))
-            //{
-            //    return new HttpMethod("CONNECT");
-            //}
-
-            throw new InvalidOperationException($"Unsupported request method '{method}'.");
-        }
+            _ => throw new InvalidOperationException($"Unsupported request method '{method}'.")
```
Comment (Kahbazi) [inline src/ReverseProxy.Core/Service/Proxy/HttpUtilities.cs:27]: @Tratcher Is there a reason YARP is not proxying custom http methods?

Comment (Tratcher) [reply src/ReverseProxy.Core/Service/Proxy/HttpUtilities.cs:27]: Probably? At the very least you need to consider the semantics of a method. E.g. we wouldn't want to proxy CONNECT using default logic, that would end badly.

**Pull request merged**